### PR TITLE
fix: solving block ssl in validate token

### PR DESCRIPTION
**Pull request by YogevBokobza**
## Description

> Describe what you did and why.

**Related issue (if any):** fixes #issue_number_goes_here

## Checklist

- [x] I have followed this repository's contributing guidelines.
- [x] I will adhere to the project's code of conduct.

## Additional information

> Anything else?

### CODE REVIEW

**Comment by YogevBokobza** [comment]
This meant to fix [this](https://github.com/home-assistant/core/pull/123578#issuecomment-2287184038)

**Comment by YogevBokobza** [comment]
I tested and looks like it's working.

**Review by TomerFi — DISMISSED** [review]

**Comment by TomerFi** [comment]
LGTM
Waiting for @thecode to have a look.

**Comment by YogevBokobza** [comment]
@thecode I would be happy if you could review it please.
Thanks.

File: src/aioswitcher/device/tools.py [hunk 1]
```diff
@@ -29,6 +30,9 @@
 
 logger = getLogger(__name__)
 
+# Preload the SSL context
+ssl_context = ssl.create_default_context()
+
```
Comment (thecode) [inline src/aioswitcher/device/tools.py]: Please move this inside the `validate_token`, I don't see any reason for making it a global variable

**Review by thecode — CHANGES_REQUESTED** [review]

File: src/aioswitcher/device/tools.py (hunk 1 above)
Comment (YogevBokobza) [reply src/aioswitcher/device/tools.py]: @thecode But the `validate_token` is an async function. Will not that result in the same issue I am trying to solve here?

Comment (thecode) [reply src/aioswitcher/device/tools.py]: https://github.com/home-assistant-libs/aiowebostv/blob/6f25f30e917d7d29d433ce03926b4571ff59b003/aiowebostv/webos_client.py#L131

An example inside an async method, it should work here also

Comment (YogevBokobza) [reply src/aioswitcher/device/tools.py]: Done.

Comment (thecode) [reply src/aioswitcher/device/tools.py]: Did you tested the change? I suggested to use `ssl_context = ssl.SSLContext()` and not `ssl.create_default_context()`

Comment (YogevBokobza) [reply src/aioswitcher/device/tools.py]: > Did you tested the change? I suggested to use `ssl_context = ssl.SSLContext()` and not `ssl.create_default_context()`

Yes I did test that and it worked, However, I changed that to what you requested.

**Review by thecode — APPROVED** [review]

**Comment by YogevBokobza** [comment]
@TomerFi Please approve and merge :)

**Review by TomerFi — APPROVED** [review]

**Comment by TomerFi** [comment]
@thecode @YogevBokobza Do we need to release?

**Comment by ybokobza** [comment]
> @thecode @YogevBokobza Do we need to release?

Yes we need.. this is been used in HA. Thanks :)

**Comment by TomerFi** [comment]
Release [4.0.3](https://github.com/TomerFi/aioswitcher/releases/tag/4.0.3).

**Pull request merged**